lib/srv/db/proxy_test.go

```diff
@@ -403,8 +403,10 @@ func TestProxyClientDisconnectDueToIdleConnection(t *testing.T) {
 	testCtx.clock.Advance(idleClientTimeout + connMonitorDisconnectTimeBuff)
 
 	waitForEvent(t, testCtx, events.ClientDisconnectCode)
-	err = mysql.Ping()
-	require.Error(t, err)
+	require.Eventually(t, func() bool {
+		err := mysql.Ping()
+		return err != nil
+	}, time.Second, 100*time.Millisecond, "failed to disconnect client conn")
```

Comment:
I would do 3-5 seconds. 1 second always seems like enough and then still fails in CI.